src/binheap.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding utf-8 -*-
+
+
+class Binheap(object):
+
+    def __init__(self, iterable=None):
+        '''Initialise a heap from an iterable or an empty heap'''
+        if iterable is None:
+            self.heap = None
+        else:
+            self.heap = sorted(iterable)
```

Comment:
A big part of this assignment is to make your heap order itself, so you can and should do without the built-in Python sorting abilities.

You've already gotten started on implementing the ordering (or heapifying, if you will) algorithm in your `push()` function, so my challenge to you is to abstract that functionality out such that it can be used in `push()`, `pop()` and `__init__()`